src/dms/core/EdFi.DataManagementService.Core/OpenApi/OpenApiDocument.cs

```diff
@@ -157,79 +157,1068 @@ private void InsertExts(
+
+    private sealed record ExtensionSchemaResult(JsonObject Schema, string? ReferencedSchemaName = null);
+
+    /// <summary>
+    /// Processes direct extension schemas (for backwards compatibility with simple extensions)
+    /// </summary>
+    private static void ProcessDirectExtensionSchema(
+        JsonObject extObjectAsObject,
+        string componentSchemaName,
+        string projectName,
+        JsonObject locationForExt,
+        JsonObject componentsSchemas
+    )
+    {
+        // Create main extension schema for the direct extension
+        string extensionSchemaName = $"{componentSchemaName}Extension";
+        string projectExtensionSchemaName = $"{projectName}_{componentSchemaName}Extension";
+
+        // Add _ext to main schema if not already there
+        if (locationForExt["_ext"] == null)
+        {
+            locationForExt.Add(
+                "_ext",
+                JsonNode.Parse($"{{ \"$ref\": \"#/components/schemas/{extensionSchemaName}\" }}")
+            );
+        }
+
+        // Create or get extension schema
+        if (!componentsSchemas.ContainsKey(extensionSchemaName))
+        {
+            componentsSchemas.Add(
+                extensionSchemaName,
+                new JsonObject { ["type"] = "object", ["properties"] = new JsonObject() }
+            );
+        }
+
+        JsonObject extensionSchema =
+            componentsSchemas[extensionSchemaName]?.AsObject()
+            ?? throw new InvalidOperationException(
+                $"Extension schema '{extensionSchemaName}' is not an object."
+            );
+
+        JsonObject mainExtensionProperties =
+            extensionSchema["properties"]?.AsObject()
+            ?? throw new InvalidOperationException(
+                $"Extension schema '{extensionSchemaName}' missing 'properties'."
+            );
+
+        // Add reference to the specific project schema
+        if (!mainExtensionProperties.ContainsKey(projectName))
+        {
+            mainExtensionProperties.Add(
+                projectName,
+                JsonNode.Parse($"{{ \"$ref\": \"#/components/schemas/{projectExtensionSchemaName}\" }}")
+            );
+        }
+
+        // Add the specific project schema with the direct extension content
+        if (!componentsSchemas.ContainsKey(projectExtensionSchemaName))
+        {
+            componentsSchemas.Add(projectExtensionSchemaName, extObjectAsObject.DeepClone());
+        }
+    }
+
+    /// <summary>
+    /// Processes a single redirected property
+    /// </summary>
+    private void ProcessRedirectedProperty(
+        string propertyName,
+        string referencedSchemaName,
+        JsonNode extensionPropertyValue,
+        JsonObject componentsSchemas,
+        string projectName
+    )
+    {
+        // Get the referenced schema
+        JsonObject? referencedSchema = componentsSchemas[referencedSchemaName]?.AsObject();
+        if (referencedSchema == null)
```

Comment:
Would something like `if (componentsSchemas[referencedSchemaName] is not JsonObject referencedSchema)`, or similar work here?